index.js

```diff
@@ -30,12 +30,18 @@ export default function stringWidth(string) {
 			continue;
 		}
 
-		// Surrogates
-		if (codePoint > 0xFFFF) {
-			index++;
+		const code = eastAsianWidth.eastAsianWidth(string.charAt(index));
+		switch (code) {
+			case 'F':
+			case 'W':
+				width += 2;
+				break;
+			case 'A':
+				width += options.ambiguousIsNarrow ? 1 : 2;
```

Comment:
`options.ambiguousIsNarrow ? 1 : 2` should be moved out of the loop, into a variable.